pkg/api/validation/validation.go

```diff
@@ -517,8 +517,18 @@ func validateAWSElasticBlockStoreVolumeSource(PD *api.AWSElasticBlockStoreVolume
 
 func validateSecretVolumeSource(secretSource *api.SecretVolumeSource, fldPath *validation.FieldPath) validation.ErrorList {
 	allErrs := validation.ErrorList{}
-	if secretSource.SecretName == "" {
+	if secretSource.SecretName == "" && len(secretSource.Items) == 0 {
 		allErrs = append(allErrs, validation.NewRequiredError(fldPath.Child("secretName")))
+		allErrs = append(allErrs, validation.NewRequiredError(fldPath.Child("items")))
+	}
+	if secretSource.SecretName != "" && len(secretSource.Items) > 0 {
```

Comment:
Look at the handling in `validateVolumeSource` and follow suit - error on all but the first one.